R/update_crop_definitions.R

```diff
@@ -0,0 +1,41 @@
+#' Update Crops Definition 
+#'
+#' @param data_frame A data frame containing the columns specified in `data_names`.
+#' @param data_names A list of column names that are needed from the dataframe.
+#' @param seasonal_data_frame A data frame containing the columns specified in `seasonal_data_names`.
+#' @param seasonal_data_names A list of column names that are needed from the `seasonal_data_frame`.
+#' @param definitions A list containing definitions to be read in.
+#' @param data_book The data book object where the data object is stored.
+#'
+#' @return A data frame in the data book at the year (and station) level containing crops definitions data.
+#' @export
+#' 
+update_crops_definitions <- function(data_frame, data_names, seasonal_data_frame, seasonal_data_names, definitions, data_book){
+    actual_crop_definitions <- definitions$crops_success
+    
+    rain_totals <- as.numeric(actual_crop_definitions$water_requirements)
+    plant_dates_from <- as.numeric(actual_crop_definitions$planting_dates$from)
+    plant_dates_to <- as.numeric(actual_crop_definitions$planting_dates$to)
+    plant_dates_by <- as.numeric(actual_crop_definitions$planting_dates$by)
+    plant_days <- c(plant_dates_from, plant_dates_to, plant_dates_by)
```

Comment:
This line here, should be

```r
plant_days <- seq(from = plant_dates_from, to = plant_dates_to, by = plant_dates_by)
```

Then we should also have this set up you've done for plant_dates for `water_requirements`, and for `planting_length` with the `from`/`to`/`by`
(If I didn't give you a from/to/by in the sample json then that's my mistake. It always is a from/to/by I believe)